lib/http-server.js

```diff
@@ -108,8 +108,8 @@ function HttpServer(options) {
       // an attacker knowledge of whether the username is correct via a timing
       // attack.
       if (credentials) {
-        var usernameEqual = secureCompare(options.username, credentials.name);
-        var passwordEqual = secureCompare(options.password, credentials.pass);
+        var usernameEqual = secureCompare(options.username.toString(), credentials.name);
+        var passwordEqual = secureCompare(options.password.toString(), credentials.pass);
```

Comment:
Should we also convert the credentials to strings? If the server were given a numeric password, as in the tests you added, the user will likely expect numeric to work. Or are we guaranteed to receive a string as credentials?

Reply:
from the [definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/HEAD/types/basic-auth/index.d.ts#L11-L17) of `basic-auth` which we use [here](https://github.com/http-party/http-server/blob/master/lib/http-server.js#L105) to parse credentials, I think the credentials are guaranteed to be string :)

Reply:
Okay, perfect, that looks like it will always be a string! Could we just add in a comment here saying the same and/or why we need to use `.toString()` here? It could be added to the comment above the `if`, or a new comment within the `if` block?

Reply:
ping @thornjad , sorry for the late replay, the comment is added, pls review again :)

Reply:
Coule we also mention in the comment that `credentials.name` and `credentials.pass` will be strings, so we don't need to convert them to strings?

Reply:
ping @thornjad , pls review again :)